br/pkg/storage/memstore.go

```diff
@@ -0,0 +1,328 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"path"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"go.uber.org/atomic"
+)
+
+type memFile struct {
+	Data atomic.Value // the atomic value is a byte slice, which can only be get/set atomically
+}
+
+// GetData gets the underlying byte slice of the atomic value
+func (f *memFile) GetData() []byte {
+	var fileData []byte
+	fileDataVal := f.Data.Load()
+	if fileDataVal != nil {
+		fileData = fileDataVal.([]byte)
+	}
+	return fileData
+}
+
+type memStorage struct {
+	rwm       sync.RWMutex
+	dataStore map[string]*memFile
+}
+
+func NewMemStorage() *memStorage {
```

Comment:
```suggestion
func NewMemStorage() *MemStorage {
```

Reply:
IMO it is unnecessary to export `memStorage` to the public, because all the exported methods are `ExternalStorage` interface methods, and there is no extra exported fields or methods.  So users typically just call the constructor and assign to an `ExternalStorage` variable.  
But considering all other implementations are exposed to the public, I'll change that.